administrator/components/com_users/src/Table/TfaTable.php

```diff
@@ -0,0 +1,458 @@
+		$query = $db->getQuery(true)
+			->update($db->quoteName('#__user_tfa'))
+			->set($db->quoteName('default') . ' = :zero')
+			->where($db->quoteName('user_id') . ' = :user_id')
+			->where($db->quoteName('id') . ' != :id')
+			->bind(':zero', $zero, ParameterType::INTEGER)
+			// phpcs:ignore
+			->bind(':user_id', $this->user_id, ParameterType::INTEGER)
+			->bind(':id', $this->id, ParameterType::INTEGER);
+		$db->setQuery($query)->execute();
+	}
+
+	/**
+	 * Regenerate backup code is the flag is set.
+	 *
+	 * @return void
+	 * @throws Exception
+	 * @since __DEPLOY_VERSION__
+	 */
+	private function generateBackupCodes(): void
+	{
+		/** @var MVCFactoryInterface $factory */
+		$factory = Factory::getApplication()->bootComponent('com_users')->getMVCFactory();
+
+		/** @var BackupcodesModel $backupCodes */
+		$backupCodes = $factory->createModel('Backupcodes', 'Administrator');
+		// phpcs:ignore
+		$user = Factory::getContainer()->get(UserFactoryInterface::class)->loadUserById($this->user_id);
+		$backupCodes->regenerateBackupCodes($user);
+	}
+
+	/**
+	 * Runs after successfully deleting a record
+	 *
+	 * @param   int|array  $pk  The promary key of the deleted record
+	 *
+	 * @return void
+	 * @since __DEPLOY_VERSION__
+	 */
+	private function afterDelete($pk): void
+	{
+		if (is_array($pk))
+		{
+			$pk = array_shift($pk);
+		}
+
+		if (!isset($this->deleteFlags[$pk]))
+		{
+			return;
+		}
+
+		if (($this->deleteFlags[$pk]['numRecords'] <= 2) && ($this->deleteFlags[$pk]['method'] != 'backupcodes'))
+		{
+			/**
+			 * This was the second to last TFA record in the database (the last one is the `backupcodes`). Therefore, we
+			 * need to delete the remaining entry and go away. We don't trigger this if the Method we are deleting was
+			 * the `backupcodes` because we might just be regenerating the backup codes.
+			 */
+			$db    = $this->getDbo();
+			$query = $db->getQuery(true)
+				->delete($db->quoteName('#__user_tfa'))
+				->where($db->quoteName('user_id') . ' = :user_id')
+				->bind(':user_id', $this->deleteFlags[$pk]['user_id'], ParameterType::INTEGER);
+			$db->setQuery($query)->execute();
+
+			unset($this->deleteFlags[$pk]);
+
+			return;
+		}
+
+		// This was the default record. Promote the next available record to default.
+		if ($this->deleteFlags[$pk]['default'])
+		{
+			$db    = $this->getDbo();
+			$query = $db->getQuery(true)
+				->select($db->quoteName('id'))
+				->from($db->quoteName('#__user_tfa'))
+				->where($db->quoteName('user_id') . ' = :user_id')
+				->bind(':user_id', $this->deleteFlags[$pk]['user_id'], ParameterType::INTEGER);
+			$ids   = $db->setQuery($query)->loadColumn();
+
+			if (empty($ids))
+			{
+				return;
+			}
+
+			$id    = array_shift($ids);
+			$one   = 1;
+			$query = $db->getQuery(true)
+				->update($db->quoteName('#__user_tfa'))
+				->set($db->quoteName('default') . ' = :one')
+				->where($db->quoteName('id') . ' = :id')
+				->bind(':one', $one, ParameterType::INTEGER)
```

Comment:
```suggestion
			$query = $db->getQuery(true)
				->update($db->quoteName('#__user_tfa'))
				->set($db->quoteName('default') . ' = 1')
				->where($db->quoteName('id') . ' = :id')
```
It does not need bind variables for values which are constant and so can't be manipulated.

Reply:
Thanks! Will merge manually.